srslog.go

```diff
@@ -31,6 +32,14 @@ func Dial(network, raddr string, priority Priority, tag string) (*Writer, error)
 	return DialWithTLSConfig(network, raddr, priority, tag, nil)
 }
 
+// DialWithCustomDialer establishes a connection by calling cdialer.
+// Each write to the returned Writer sends a log message with the given facility, severity and tag.
+// While network and raddr will be passed to cdialer, it is allowed for cdialer to ignore them.
+// If cdialer is nil, this function behaves like Dial.
+func DialWithCustomDialer(network, raddr string, priority Priority, tag string, cdialer func(string, string) (net.Conn, error)) (*Writer, error) {
```

Comment:
We're repeating `func(string, string) (net.Conn, error)` a few times; are we better served by adding

```go
type DialFunc func(string, string) (net.Conn, error)
```

so we don't clutter up these function signatures?

Reply:
I considered doing that, was just trying to add as few lines/types as possible. I don't have a problem with adding that.

Reply:
I agree with you about not wanting to add types, if possible. Does Go's `net` package specify an interface anywhere that defines this dialer signature? (Seems like the kind of thing it might.)